files/scripts/healdata/heal-cedar-data-ingest.py

```diff
@@ -79,53 +79,82 @@ def update_filter_metadata(metadata_to_update):
-            mds_discovery_data_body = {}
-            if mds_res["_guid_type"] == "discovery_metadata":
-                print("Metadata is already registered. Updating MDS record")
-            elif mds_res["_guid_type"] == "unregistered_discovery_metadata":
-                print("Metadata is has not been registered. Registering it in MDS record")
+limit = 10
+offset = 0
+
+# initalize this to be bigger than our inital call so we can go through while loop
+total = 100
+
+while((limit + offset <= total)):
+    # Get the metadata from cedar to register
+    print("Querying CEDAR...")
+    cedar = requests.get(f"http://revproxy-service/cedar/get-instance-by-directory/{dir_id}?limit={limit}&offset={offset}", headers=token_header)
+
+    # If we get metadata back now register with MDS
+    if cedar.status_code == 200:
+        metadata_return = cedar.json()
+        if "metadata" not in metadata_return:
+            print("Got 200 from CEDAR wrapper but no metadata in body, something is not right!")
+            sys.exit(1)
+
+        total = metadata_return["metadata"]["totalCount"]
+        returned_records = len(metadata_return["metadata"]["records"])
+        print(f"Successfully got {returned_records} record(s) from CEDAR directory")
+        for cedar_record in metadata_return["metadata"]["records"]:
+            if "appl_id" not in cedar_record:
+                print("This record doesn't have appl_id, skipping...")
                 continue
-            pydash.merge(mds_discovery_data_body, mds_res["gen3_discovery"], cedar_record)
-            mds_discovery_data_body = update_filter_metadata(mds_discovery_data_body)
-            mds_cedar_register_data_body["gen3_discovery"] = mds_discovery_data_body
-            mds_cedar_register_data_body["_guid_type"] = "discovery_metadata"
-
-            print("Metadata is now being registered.")
-            mds_put = requests.put(f"http://revproxy-service/mds/metadata/{cedar_record_id}",
-                headers=token_header,
-                json = mds_cedar_register_data_body
-            )
-            if mds_put.status_code == 200:
-                print(f"Successfully registered: {cedar_record_id}")
+
+            # get the appl id from cedar for querying in our MDS
+            cedar_appl_id = str(cedar_record["appl_id"])
+
+            # Get the metadata record for the nih_application_id
+            mds = requests.get(f"http://revproxy-service/mds/metadata?gen3_discovery.appl_id={cedar_appl_id}&data=true")
+            if mds.status_code == 200:
+                mds_res = mds.json()
+
+                # the query result key is the record of the metadata. If it doesn't return anything then our query failed.
+                if len(list(mds_res.keys())) == 0:
+                    print("Query returned nothing for ", cedar_appl_id, "appl id")
+                    continue
+
+                # get the key for our mds record
+                cedar_record_id = list(mds_res.keys())[0]
```

Comment:
better also check `if len(list(mds_res.keys())) > 1` before doing this just for sanity check. It shouldn't happen, but if it does, then it probably means we have some issue with our metadata and we should investigate